### PR TITLE
chore(v0.50.0): Migrating codebase to latest SDK version

### PR DESCRIPTION
**Pull request by davidterpay**
### Overview
This PR does all of the basic migration of POB V1 to V2 using sdk version `v0.50.0`. Outside of basic import changes, the only major change is the expected keepers interface for some of the keepers.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,7 +70,20 @@ $(BUILD_TARGETS): $(BUILD_DIR)/
 $(BUILD_DIR)/:
 	mkdir -p $(BUILD_DIR)/
 
-.PHONY: build-test-app
+# build-and-start-app builds a slinky simulation application binary in the build folder
```
Comment (nivasan1) [inline Makefile:73]: nit: slinky

File: x/builder/types/params.go [hunk 2]
```diff
@@ -49,15 +49,20 @@ func DefaultParams() Params {
 
 // Validate performs basic validation on the parameters.
 func (p Params) Validate() error {
+	if p.EscrowAccountAddress == nil {
```
Comment (davidterpay) [inline x/builder/types/params.go:52]: Small update to how we validate params

File: x/builder/types/codec.go [hunk 3]
```diff
@@ -7,27 +7,17 @@ import (
...
 var (
 	amino     = codec.NewLegacyAmino()
-	ModuleCdc = codec.NewAminoCodec(amino)
+	ModuleCdc = codec.NewLegacyAmino()
 )
 
 func init() {
 	RegisterLegacyAminoCodec(amino)
 	cryptocodec.RegisterCrypto(amino)
 	sdk.RegisterLegacyAminoCodec(amino)
```
Comment (davidterpay) [inline x/builder/types/codec.go:20]: deprecating this since sdk version no longer supports this

File: x/builder/types/params.go (hunk 2 above)
Comment (nivasan1) [reply x/builder/types/params.go:52]: 👍

File: x/builder/types/codec.go (hunk 3 above)
Comment (nivasan1) [reply x/builder/types/codec.go:20]: 👍

File: blockbuster/abci/check_tx.go [hunk 4]
```diff
@@ -127,24 +134,36 @@ func (handler *CheckTxHandler) CheckTx() CheckTx {
 		// We attempt to get the latest committed state in order to verify transactions
 		// as if they were to be executed at the top of the block. After verification, this
 		// context will be discarded and will not apply any state changes.
-		ctx := handler.GetContextForBidTx(req)
+		ctx, _ := handler.baseApp.GetContextForFinalizeBlock(req.Tx).CacheContext()
```
Comment (nivasan1) [inline blockbuster/abci/check_tx.go:137]: We don't need to set CheckTx mode here? I.e IsCheckTx / IsRecheckTx?

**Review by nivasan1 — APPROVED** [review]
Left comments + questions. PR is p large so lmk if there's anything you'd like reviewed in more depth.

**Pull request merged**